lib/macho.rb

```diff
@@ -39,4 +40,22 @@ def self.open(filename)
 
     file
   end
+
+  # Signs the dylib using an ad-hoc identity.
+  # Necessary after making any changes to a dylib, since otherwise
+  # changing a signed file invalidates its signature.
+  # @param filename [String] the file being opened
+  # @return [void]
+  # @raise [ModificationError] if the operation fails
+  def self.codesign(filename)
```

Comment:
```suggestion
  def self.codesign!(filename)
```

maybe?

Reply:
Yep, this LGTM: it's consistent with the other methods in ruby-macho that actually modify a file in-place.